jvb/src/main/kotlin/org/jitsi/videobridge/cc/vp9/Vp9AdaptiveSourceProjectionContext.kt

```diff
@@ -334,6 +335,49 @@ class Vp9AdaptiveSourceProjectionContext(
         )
     }
 
+    /**
+     * Create a projection for the first frame after a resumption, i.e. when a source is turned back on.
+     */
+    private fun createResumptionProjection(
+        frame: Vp9Frame,
+        initialPacket: Vp9Packet,
+        mark: Boolean,
+        receivedMs: Long
+    ): Vp9FrameProjection {
+        lastPicIdIndexResumption = frame.index
+
+        /* These must be non-null because we don't execute this function unless
+            frameIsNewSsrc has returned false.
+        */
+        val lastFrame = prevFrame(frame)!!
+        val lastProjectedFrame = lastVp9FrameProjection.vp9Frame!!
+
+        /* Project timestamps linearly. */
+        val tsDelta = getTimestampDiff(
+            lastVp9FrameProjection.timestamp,
+            lastProjectedFrame.timestamp
+        )
+        val projectedTs = applyTimestampDelta(frame.timestamp, tsDelta)
+
+        /* Increment picId and tl0picidx by 1 from the last projected frame. */
+        val projectedPicId = applyExtendedPictureIdDelta(lastVp9FrameProjection.pictureId, 1)
+        val projectedTl0PicIdx = applyTl0PicIdxDelta(lastVp9FrameProjection.tl0PICIDX, 1)
+
+        /* Increment sequence numbers based on the last projected frame, but leave a gap
```

Comment:
how does that work? You can see if the current packet is not the beginning of the frame (i.e. B is not set), but you can't know how many packets of the new frame are missing?

I'd expect more than one missing/reordered packet to be quite rare though.

Reply:
I'm keeping track of all received frames, not just those in frames that I've forwarded.  Thus I can calculate the sequence number gap between the most recent packet previously received and this packet, and so leave the equivalent gap in the outgoing sequence number space.